README.md

```diff
@@ -2,31 +2,242 @@
+      System.out.println(greeting3.hashCode());
+   }
+}
+```
+
+An output for the above code may look like this:
+
+```text
+-862545276
+-862545276
+-1397214398
+```
+
+It should be noted that the hashed values here may differ from the hashed
+values if you were to run this code on your machine.
+
+### `equals()`
+
+The Java `Object` class also has an `equals()` method to help us compare if
+two objects are "equal". It will take in another object as a parameter and
+compare it to the calling object. If the objects are equal to each other, then
+the method will return true, otherwise it will return false. For example:
+
+```java
+public class Example {
+    public static void main(String[] args) {
+        String greeting1 = "Hello World";
+        String greeting2 = "Hello World";
+        String greeting3 = "Welcome";
+
+        System.out.println(greeting1.equals(greeting2));
+        System.out.println(greeting2.equals(greeting3));
+    }
+}
+```
+
+If we were to run the code above, the result would look like this:
+
+```text
+true
+false
+```
+
+It should also be noted that when comparing `String` values that we will want
+to use the `equals()` method as the `==` only works on numbers.
+
+It is highly recommended to override the `equals()` method and the `hashCode()`
+method together when we create our own classes to ensure that the objects are
+equal in a logical sense. For example, if we had a `Bicycle` class, we would
+want to override the `equals()` and `hashCode()` methods to ensure that a red
+bike would be indeed equal to another red bike.
+
+```java
+public class Bicycle {
+    private String color;
+    
+    Bicycle(String color) {
+        this.color = color;
+    }
+    
+    @Override
+    public boolean equals(Bicycle bicycle) {
+        return this.color.equals(bicycle.color);
+    }
+    
+    @Override
+    public int hashCode() {
+        return this.color.hashCode();
+    }
+}
+```
+
+### `toString()`
+
+The `toString()` method returns a `String` representation of the instance and is
+used to convert the object to a `String`.
+
+If the `toString()` method is not overridden, then the default `String` that
+will be returned. The default `String` will have the following syntax
+
+```text
+getClass().getName() + '@' + Integer.toHexString(hashCode())
+// Bicycle@d716361
+```
+
+The first part of the default `String` is the name of the class followed by the
+`@` symbol. The last part of the default `String` is a hexadecimal value of the
+hash code. Hexadecimal is a different number system that uses a base of 16 rather
+than 10, which is the decimal system, and uses characters 0-9 and a-f. It should
+be noted again, that the default `String` of `className@hashCode` is **not**
+returning the address in memory. It is a common misconception that the hashcode
+returns a point in memory.
+
+```java
+public class Bicycle {
+   public static void main(String[] args) {
+      Bicycle bicycle = new Bicycle();
```

Comment:
Did you define an empty constructor for Bicycle?

Reply:
Since no constructor is defined, the default no-args constructor should be generated.

Reply:
Line 148 has a constructor with a color parameter